app-games/src/main/java/com/aptoide/android/aptoidegames/feature_apps/presentation/BonusViewModel.kt

```diff
@@ -0,0 +1,48 @@
+package com.aptoide.android.aptoidegames.feature_apps.presentation
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import cm.aptoide.pt.extensions.runPreviewable
+import com.aptoide.android.aptoidegames.feature_apps.data.BonusData
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.flow.update
+import javax.inject.Inject
+
+@HiltViewModel
+class BonusViewModel @Inject constructor() : ViewModel() {
+
+  private val viewModelState = MutableStateFlow<Pair<String,String>?>(null)
+
+  val uiState = viewModelState
+    .stateIn(
+      viewModelScope,
+      SharingStarted.Eagerly,
+      viewModelState.value
+    )
+
+  init {
+    viewModelState.update { BonusData.getBonusData() }
+  }
+
+  fun addBundleData(title: String, tag: String) {
+    BonusData.setBonusData(title, tag)
+    viewModelState.update { title to tag }
+  }
+}
+
+@Composable
+fun rememberBonusBundle() = runPreviewable(
+  preview = { null },
+  real = {
+    val vm = hiltViewModel<BonusViewModel>()
+    val bonusInfo by vm.uiState.collectAsState()
+    bonusInfo
```

Comment:
Make a method encapsulation here for wider usage
`bonusInfo to bonusViewModel.addBundleData`